core/src/main/kotlin/fr/sncf/osrd/api/etcs/ETCSBrakingCurvesEndpoint.kt

```diff
@@ -0,0 +1,197 @@
+                    powerRestrictionsLegacyMap,
+                    rollingStock.powerRestrictions,
+                    request.useElectricalProfiles
+                )
+            val curvesAndConditions =
+                rollingStock.mapTractiveEffortCurves(electrificationMap, request.comfort)
+            val signalingRanges = buildSignalingRanges(infra, blockPath, chunkPath)
+            val stopPositions = getStopPositions(parseRawSimulationScheduleItems(request.schedule))
+            val context =
+                EnvelopeSimContext(
+                    rollingStock,
+                    envelopeSimPath,
+                    2.0,
+                    curvesAndConditions.curves,
+                    makeETCSContext(rollingStock, infra, chunkPath, routePath, signalingRanges)
+                )
+
+            // Parse mrsp.
+            val mrsp = parseRawMrsp(request.mrsp, Offset(envelopeSimPath.length.meters))
+
+            // Compute ETCS braking curves.
+            val etcsSimulator = ETCSBrakingSimulatorImpl(context)
+            // Compute slowdown braking curves.
+            val slowdowns = etcsSimulator.computeSlowdownLocations(mrsp)
+            val slowdownBrakingCurves = etcsSimulator.computeSlowdownBrakingCurves(mrsp, slowdowns)
+            // Compute stop braking curves.
+            val stops = stopPositions.map { Offset<TravelledPath>(it.meters) }
+            val etcsStops = etcsSimulator.computeStopLocations(mrsp, stops)
+            val stopBrakingCurves = etcsSimulator.computeStopBrakingCurves(mrsp, etcsStops)
+            // Compute block transition braking curves.
+            val etcsSignalBlockPathOffsets =
+                infra.blockInfra.getSignalOffsets(infra.rawInfra, blockPath, ETCS_LEVEL2.id)
+            val travelledPathOffset =
+                Distance.ZERO -
+                    trainPathBlockOffset(infra.rawInfra, infra.blockInfra, blockPath, chunkPath)
+                        .distance
+            val etcsSignalOffsets =
+                etcsSignalBlockPathOffsets
+                    .map { Offset<TravelledPath>(it.distance - travelledPathOffset) }
+                    .filter { it.distance > Distance.ZERO && it.distance <= chunkPath.length }
+            val etcsSignalEoas = etcsSimulator.computeStopLocations(mrsp, etcsSignalOffsets)
+            val signalBrakingCurves = etcsSimulator.computeStopBrakingCurves(mrsp, etcsSignalEoas)
+
+            // Parse response.
+            val res =
+                ETCSBrakingCurvesResponse(
+                    slowdownBrakingCurves.map { parseETCSCurves(it.value) },
+                    stopBrakingCurves.map { parseETCSCurves(it.value) },
+                    signalBrakingCurves.map { parseETCSCurves(it.value) }
+                )
+            RsJson(RsWithBody(etcsBrakingCurvesResponseAdapter.toJson(res)))
+        } catch (ex: Throwable) {
+            ExceptionHandler.handle(ex)
+        }
+    }
+
+    private fun parseRawMrsp(
```

Comment:
The implementation is a bit complicated and hard to review.

We could implement a dedicated method to convert `RangeValues` into `DistanceRangeMap` (I'm surprized there isn't one already), it would be useful and make this method much simpler.

You could also use `MRSPEnvelopeBuilder`

(We've had bugs in similar code before, and endpoint parsing is tedious to test)